modules/api/php/models/projectinstrumentsrow.class.inc

```diff
@@ -34,17 +34,13 @@ class ProjectInstrumentsRow implements \LORIS\Data\DataInstance
      */
     public function __construct(array $row)
     {
-        $shortname      = $row['shortname'] ?? null;
-        $ddeinstruments = array_keys(\Utility::getAllDDEInstruments());
-
-        $this->_shortname    = $shortname;
+        $this->_shortname    = $row['shortname'] ?? null;
         $this->_fullname     = $row['fullname'] ?? null;
         $this->_subgroupname = $row['subgroupname'] ?? null;
-        $this->_isDDE        = in_array(
-            $shortname,
-            $ddeinstruments,
-            true
```

Comment:
This is not a backwards compatible change in the API and can not be done on the released version of the API. (If changed in the dev version, the swagger schema and markdown API documentation also need to be updated for them.)

Reply:
hmm I see. I am not sure what to do about this then as the isDDE field doesn't really mean anything anymore once the DDE is based on the Visit/Instrument tuple instead of just the instrument. 

Are you saying I should update the swagger schema and markdown API documentation, or that I should undo the changes since it is not backwards compatible?

Alternatively, I could add ddeVisits as I have done, and also keep isDDE to return true only if the instrument has DDE for all visits that it is in?

Reply:
I think for v0.0.3, I think you should have isDDE return true if the instrument has DDE for any visits.

For v0.0.4-dev, I think you should add ddeVisits as you've done but also update the swagger schema and markdown documentation.